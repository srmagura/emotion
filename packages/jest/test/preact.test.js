/* eslint-disable react/no-unknown-property */
import 'test-utils/legacy-env'
/** @jsx h */
import { h } from 'preact'
import render from 'preact-render-to-string'
import prettyFormat from 'pretty-format'
import * as emotion from '@emotion/css'
import { createSerializer } from '@emotion/jest'
const { ReactElement, ReactTestComponent, DOMElement } = prettyFormat.plugins

let emotionPlugin = createSerializer()

describe('jest-emotion with preact', () => {
  const divStyle = emotion.css`
    color: red;
  `
  const svgStyle = emotion.css`
    width: 100%;
  `

  it('replaces class names and inserts styles into preact test component snapshots', () => {
    const tree = render(
      <div class={divStyle}>
        <svg class={svgStyle} />
      </div>
    )

    expect(tree).toMatchSnapshot()
  })

  it('handles elements with no props', () => {
    const tree = render(<div />)

    expect(tree).toMatchSnapshot()
  })
})
