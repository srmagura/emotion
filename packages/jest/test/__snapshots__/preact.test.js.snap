// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-emotion with preact handles elements with no props 1`] = `"<div></div>"`;

exports[`jest-emotion with preact replaces class names and inserts styles into preact test component snapshots 1`] = `"<div class=\\"css-17f3slo\\"><svg class=\\"css-ybrhvy\\"></svg></div>"`;
